### PR TITLE
feat(ui): Add post screen

### LINKED ISSUE #63: Create the post screen

### PR DESCRIPTION
**Pull request by MarcoLourenco55**
**Description**
This PR introduces a new Post screen that lets people read a post, follow the conversation, and participate easily.
It introduces the following features:
* Readable post view: clear header with author and time, title, body, and tag chips in one card
* Threaded comments: replies are shown in a collapsible thread, making long discussions easier to follow
* Inline replying: reply directly under a comment with a quick-send action
* Top-level commenting: composer fixed at the bottom to share new thoughts anytime
* Self-management: users can delete their own posts and comments
* Clean visuals: subtle gutters indicate depth; placeholders for avatars; unknown authors are clearly labeled

**Visuals**
* Main screen
Image  PostScreen

* Comment and subcomments
Screenshot 2025-10-28 140427

**Testing**
* Number of tests: 13
* Running time: ~46 seconds (locally)
* Line coverage: ~94%
Coverage  PostScreen_JacocoReport

**References**
* Issues: https://github.com/Meeple-Meet/Meeple-Meet/issues/63

**TODOS**
* Integrate it in the navigation (waiting for the necessary screens to be integrated first)

(This text was reformulated by ChatGPT-5 Thinking)

### CODE REVIEW

File: app/src/androidTest/java/com/github/meeplemeet/ui/PostScreenTest.kt [hunk 1]
```diff
@@ -0,0 +1,618 @@
...
+  /* Canonical accounts for examples */
+  private val marco =
+      Account(uid = "u_marco", handle = "marco", name = "Marco", email = "marco@meeple.ch")
+  private val alex =
+      Account(uid = "u_alex", handle = "alex", name = "Alex", email = "alex@meeple.ch")
+  private val dany =
+      Account(uid = "u_dany", handle = "dany", name = "Dany", email = "dany@meeple.ch")
+  private val ts = Timestamp(1_725_000_000, 0)
+
+  private fun c(id: String, text: String, author: Account, vararg children: Comment) =
```
Comment (dtbcds) [inline app/src/androidTest/java/com/github/meeplemeet/ui/PostScreenTest.kt]: Consider renaming this function to something a bit more explicit, it's hard to tell what it does without looking at its body.

File: app/src/androidTest/java/com/github/meeplemeet/ui/PostScreenTest.kt [hunk 2]
```diff
@@ -0,0 +1,618 @@
...
+            postId = postId,
+            postViewModel = postVM,
+            usersViewModel = usersVM,
+            onBack = onBack)
+      }
+    }
+  }
+
+  /* Query helpers */
+  private fun n(tag: String) = compose.onNodeWithTag(tag, useUnmergedTree = true)
```
Comment (dtbcds) [inline app/src/androidTest/java/com/github/meeplemeet/ui/PostScreenTest.kt]: Same as above

File: app/src/androidTest/java/com/github/meeplemeet/ui/PostScreenTest.kt [hunk 3]
```diff
@@ -0,0 +1,618 @@
...
+
+  private fun t(text: String, unmerged: Boolean = true) =
```
Comment (dtbcds) [inline app/src/androidTest/java/com/github/meeplemeet/ui/PostScreenTest.kt]: Same as above

File: app/src/main/java/com/github/meeplemeet/ui/PostScreen.kt [hunk 4]
```diff
@@ -0,0 +1,916 @@
...
+/**
+ * A custom layout that arranges its children in a horizontal flow, wrapping to new lines as needed.
+ *
+ * @param modifier Modifier to be applied to the layout.
+ * @param horizontalGap Horizontal gap between items.
+ * @param verticalGap Vertical gap between rows.
+ * @param content Composable content to be laid out.
+ */
+@Composable
+private fun FlowRow(
```
Comment (dtbcds) [inline app/src/main/java/com/github/meeplemeet/ui/PostScreen.kt]: I'm not sure if the FlowRow provided by android compose does the same thing, but if it does, consider using it instead of making a custom one for less maintenance and more edge case handling.

**Review by dtbcds — CHANGES_REQUESTED** [review]
Overall, really good PR. Please make the helper functions for the tests more readable and do look into the android compose layout FlowRow.

File: app/src/main/java/com/github/meeplemeet/ui/PostScreen.kt (hunk 4 above)
Comment (MarcoLourenco55) [reply app/src/main/java/com/github/meeplemeet/ui/PostScreen.kt]: Thank you for the info, in fact this is way cleaner !
Solved in commit https://github.com/Meeple-Meet/Meeple-Meet/pull/70/commits/490175b0219787456a4fd0064b1bcea729fb8345

File: app/src/androidTest/java/com/github/meeplemeet/ui/PostScreenTest.kt (hunk 1 above)
Comment (MarcoLourenco55) [reply app/src/androidTest/java/com/github/meeplemeet/ui/PostScreenTest.kt]: Solved in commit https://github.com/Meeple-Meet/Meeple-Meet/pull/70/commits/bef3c8f0add47673fb59c7cc9f1be3dcb7cdebea

File: app/src/androidTest/java/com/github/meeplemeet/ui/PostScreenTest.kt (hunk 3 above)
Comment (MarcoLourenco55) [reply app/src/androidTest/java/com/github/meeplemeet/ui/PostScreenTest.kt]: Solved in commit https://github.com/Meeple-Meet/Meeple-Meet/pull/70/commits/bef3c8f0add47673fb59c7cc9f1be3dcb7cdebea

**Review by dtbcds — APPROVED** [review]
Looks good to me after the changes!
